packages/iov-bns/src/encode.spec.ts

```diff
@@ -57,10 +58,49 @@ describe("Encode", () => {
     expect(Uint8Array.from(encoded)).toEqual(privBin);
   });
 
-  it("encodes amount", () => {
-    const amount = encodeAmount(coinJson);
-    const encoded = codecImpl.x.Coin.encode(amount).finish();
-    expect(Uint8Array.from(encoded)).toEqual(coinBin);
+  describe("encodeAmount", () => {
+    it("can encode amount 3.123456789 ASH", () => {
+      const amount: Amount = {
+        quantity: "3123456789",
```

Comment:
This is where the `toQuantity` helper I proposed above would look nice... it would just match the description.

Reply:
Counter offer (as I wrote above): `quantity: "3_123456789"`